ENIGMAsystem/SHELL/Graphics_Systems/General/GSmatrix.h

```diff
@@ -60,13 +98,6 @@ void d3d_transform_set_rotation_y(gs_scalar angle);
 void d3d_transform_set_rotation_z(gs_scalar angle);
 void d3d_transform_set_rotation_axis(gs_scalar x, gs_scalar y, gs_scalar z, gs_scalar angle);
 void d3d_transform_set_rotation(gs_scalar x, gs_scalar y, gs_scalar z);
-void d3d_transform_set_look_at(gs_scalar xfrom, gs_scalar yfrom, gs_scalar zfrom,gs_scalar xto, gs_scalar yto, gs_scalar zto, gs_scalar xup, gs_scalar yup, gs_scalar zup);
```

Comment:
Actually, I removed this function and did not bring it back as I stated above. To answer your question, the function was not idempotent though. I have several reasons to argue for not bringing this back.

* It's still possible to do this with the `matrix_` functions
* You don't want to apply a view matrix to the world matrix, because that's not how GM's API is designed
* The function `d3d_set_projection` does this, but with the correct matrix, the view matrix
* Allowing this function makes our lighting more difficult because then we would have to split the look at off of the world matrix in order to set the light positions
* The transform functions were mainly intended by Mark Overmars for the user to transform objects, not projections, that's what the projection functions are for

Considering all of that, I think you can see that `d3d_transform_set_look_at` and `d3d_transform_add_look_at` are a bad idea.